DevEdu.API/DevEdu.Business/Services/TaskService.cs

```diff
@@ -9,39 +13,59 @@ public class TaskService : ITaskService
         private readonly ITaskRepository _taskRepository;
         private readonly ICourseRepository _courseRepository;
         private readonly IStudentAnswerOnTaskRepository _studentAnswerOnTaskRepository;
+        private readonly ITaskValidationHelper _taskValidationHelper;
 
-        public TaskService(ITaskRepository taskRepository, ICourseRepository courseRepository, IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository)
+        public TaskService(
+            ITaskRepository taskRepository,
+            ICourseRepository courseRepository,
+            IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository, 
+            ITaskValidationHelper taskValidationHelper
+        )
         {
             _taskRepository = taskRepository;
             _courseRepository = courseRepository;
             _studentAnswerOnTaskRepository = studentAnswerOnTaskRepository;
+            _taskValidationHelper = taskValidationHelper;
         }
 
-        public TaskDto GetTaskById(int id) => _taskRepository.GetTaskById(id);
-
-        public TaskDto GetTaskWithCoursesById(int id)
+        public TaskDto GetTaskById(int id)
         {
+            // check if task exists
+            _taskValidationHelper.CheckTaskExistence(id);
             var taskDto = _taskRepository.GetTaskById(id);
-            taskDto.Courses = _courseRepository.GetCoursesToTaskByTaskId(id);
             return taskDto;
         }
 
-        public TaskDto GetTaskWithAnswersById(int id)
+        public TaskDto GetTaskByIdWithValidation(int id, int userId)
         {
-            var taskDto = _taskRepository.GetTaskById(id);
-            taskDto.StudentAnswers = _studentAnswerOnTaskRepository.GetStudentAnswersToTaskByTaskId(id);
+            _taskValidationHelper.CheckTaskExistenceWithValidation(id, userId);
```

Comment:
1. check if user exists, and then...
2. check if task exists and modify the method for it to return a dto
3. and only after this make a call to CheckTaskExistenceWithValidation
4. return the dto received in step 2

on step 3, if we found that user doesn't have rights to get this task, we will throw an error
and in this case the dto won't be returned from step 4 
so it's ok to have such implementation